ooniprobe/View/TestResults/Footer/UploadFooterViewController.m

```diff
@@ -0,0 +1,124 @@
+#import "UploadFooterViewController.h"
```

Comment:
This file makes me a little bit sad. It seems that it contains both UI handling and code for resubmitting measurements. I would be happier if we could have a clean separation of these two concerns.

I'd like the code for resubmitting to be totally independent from UI, so that we can (1) submit it as a separate PR without wondering about the storyboard and (2) write some tests for it. The latter will allow us to be more confident that the code that we're merging into master works as intended.

Reply:
The downside of having two classes is not having full control over UI events. As the upload is used only in the footer and the footer is a reusable class in any view it's best to keep most of the code here. The only function that can be moved away is `uploadSingleMeasurement` bu it will decrease the readability

Reply:
So basically this is not a real UI class but a detachable piece of UI

Reply:
Okay, I'll need to think a little bit. My main concern is that I want some pieces of code to be testable and so I need to understand how to proceed here.

Reply:
Eventually the upload functions are detachable, but
- There will be much code in general for the back and forth communication with UI
- One more class to manage uploaded used by only one class
- The `UploadFooterViewController` class will become only its `viewDidLoad` method

Reply:
I really want to pursue this goal of separating. If not now, maybe at a later time.